### PR TITLE
Don't trigger the return_value_from_void_function warning from initializers

### PR DESCRIPTION
**Pull request by mrbkap**
Consider the code:
```swift
func foo() {
   enum Bar: RawRepresentable() {
       case one
       case two

       init?(rawValue: String) {
           return nil
       }
   }
}
```

This example should not trigger `return_value_from_void_function` in the initializer because the initializer is nullable and can return `nil`. However, using the latest version of `SwiftLint` I'm seeing this rule trigger on this example.

The problem appears to be that `Syntax.enclosingFunction` extension climbs the AST's parent chain looking for a `FunctionDecl` but initializers are neither blocks nor variable declarations. By bailing when we see an initializer, we won't trigger this rule incorrectly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -130,6 +130,10 @@
   [SimplyDanny](https://github.com/SimplyDanny)
   [#3852](https://github.com/realm/SwiftLint/pull/3852)
 
+* Don't trigger the return_value_from_void_function warning from initializers.
```
Comment (SimplyDanny) [inline CHANGELOG.md:133]:
```suggestion
* Don't trigger the `return_value_from_void_function` warning from initializers.  
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -130,6 +130,10 @@
   [SimplyDanny](https://github.com/SimplyDanny)
   [#3852](https://github.com/realm/SwiftLint/pull/3852)
 
+* Don't trigger the return_value_from_void_function warning from initializers.  
+  [mrbkap](https://github.com/mrbkap)
+  [#5429](https://github.com/realm/SwiftLint/pull/5429)
```
Comment (SimplyDanny) [inline CHANGELOG.md:135]:
```suggestion
  [mrbkap](https://github.com/mrbkap)
```

File: Source/SwiftLintBuiltInRules/Rules/Idiomatic/ReturnValueFromVoidFunctionRuleExamples.swift [hunk 3]
```diff
@@ -45,6 +45,17 @@ internal struct ReturnValueFromVoidFunctionRuleExamples {
...
+        func baz() {
+            enum Foo {
+                case bar
+
+                init?(arg: String?) {
+                    return nil
+                }
+            }
+        }
+        """),
```
Comment (SimplyDanny) [inline Source/SwiftLintBuiltInRules/Rules/Idiomatic/ReturnValueFromVoidFunctionRuleExamples.swift:57]:
```suggestion
        """, excludeFromDocumentation: true),
```

**Review by SimplyDanny — APPROVED** [review]
Looks good to me. Thanks!

**Comment by mrbkap** [comment]
Thanks for the quick action and cleanups!

**Pull request merged**